platforms/core-runtime/daemon-protocol/src/main/java/org/gradle/launcher/daemon/configuration/DaemonPriority.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher.daemon.configuration;
+
+public enum DaemonPriority {
```

Comment:
This feels like a good time to add javadoc here. Now that its in its own file, it loses the context of the file it used to live in. 

What does it mean for a daemon to have a priority? How does a daemon with a higher priority interact with a daemon with a lower priority?

Reply:
The goal for this PR is simply to move things around, not improve them. It was called `DaemonParameters.Priority` before, now it's called `DaemonPriority`, with the same amount of Javadoc. Could it be better? yes. Is it worse than it was before? no.